### DISCUSSION TITLE
multiple forms are somehow connected

### DISCUSSION THREAD

**Discussion by Charlotte-Ketteler** [neuefische/questions#664] [Web] [answered]
I am building a booking app where the user can book a riding lesson for the day. 
Right now I am fixing my BookingPage(because of uplifting) where there are 10 buttons displayed showing the timeslots for the riding lessons. When clicking on a timeslot a form opens an the user can choose the amount of participants and the horses they want to ride. Now the issue is, that every form of every timeslot is active. For example I am choosing 4 participants for 8:00 am. All other timeslots will now have 4 participants selected. The same with the horses. I am sure it has something to do with my TimeSlotsButton and BookingForm component but I can't figure out how to solve it. Maybe you have an idea? My branch is called  ["Terminübersicht".](https://github.com/Charlotte-Ketteler/capstone-project/tree/Termin%C3%BCbersicht). This is my [US1](https://github.com/Charlotte-Ketteler/capstone-project/issues/1) so you get an understanding of how it should look like. Please note that for the horses I made buttons instead of a select box.

**Comment by florianblaschke** [comment, marked as answer]
Hi Charlotte!

Your forms are connected because they all are dependant on your  const [showForm, setShowForm] = useState(false);

With this, you manage the showForm-State for all Forms. So, if you click on one -> everything will open because showForm is true and thats what you ask here: {showForm && (
        <BookingForm etc...

You will need to find a way to target each form separately.

You could use your already existent showForm. But this need to have a state for every single form. And your function needs to look something like this: `

const test = [
  { id: 1, hide: false },
  { id: 2, hide: false },
  { id: 3, hide: false },
  { id: 4, hide: false },
];

export default function Test() {
  const [showForm, setShowForm] = useState(test);

  function toggleForm(id) {
    setShowForm(
      showForm.map((slot) => {
        if (slot.id === id) return { ...slot, hide: !slot.hide };
        return slot;
      })
    );
  }

  return <button onClick={() => toggleForm(id)}>SETFORM</button>
}
`
Also you have to place the state management into your app.js and pass (in my example "hide") it down.

Hope that helps!
